xds/internal/httpfilter/fault/fault_test.go

```diff
@@ -489,6 +482,15 @@ func (s) TestFaultInjection_Unary(t *testing.T) {
 				return time.NewTimer(0)
 			}
 
+			serviceName := fmt.Sprintf("myservice%d", tcNum)
```

Comment:
I think we should be able to the same resource if we have a move the two lines I mentioned in the other comment to be inside of `t.Run()`.

Reply:
I tried this, actually, but still had issues due to a race since the xdsclient is global.